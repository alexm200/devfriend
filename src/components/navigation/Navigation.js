import React, { Component } from 'react';
import Logo from './compontents/Logo';
import NavigationItem from './compontents/NavigationItem';
import navStyles from '../../styles/modules/navigation.module.scss';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { uiActions } from '../../store/actions/ui';
import { menuItemActions } from '../../store/actions/menuItem';
import { library } from '@fortawesome/fontawesome-svg-core';

class Navigation extends Component {

  render() {  
    return (
        <ul className={[navStyles.sidebar, "navbar-nav"].join(" ")}>
            <Logo></Logo>

            <div style={{ display: this.props.user.isLoggedIn ? "block" : "none" }}>

              <hr className={navStyles.divider}></hr>
              <NavigationItem icon="home" text="Home" navigateUrl="/"></NavigationItem>
              <hr className={navStyles.divider}></hr>
              {
                [...this.props.menuItems].sort((x, y) => { return y.order - x.order }).map((i, index) => {               
                  return (
                    <div key={index}>
                      {
                        i.isHeader ?
                        <div className={navStyles.heading}>{i.text}</div>
                        :
                        <div>
                          <NavigationItem icon={Object.keys(library.definitions.fab).includes(i.icon) ? ['fab', i.icon] : i.icon} text={i.text} navigateUrl={`/_${i.text}`}></NavigationItem>
                          {i.hasDivider && <hr className={navStyles.divider}></hr>}
                        </div>
                      }                    
                    </div>                  
                  )                
                })
              }

              <NavigationItem icon="edit" text="Edit Menu" class={navStyles.addBtn} navigateUrl="/menu"></NavigationItem>

            </div>

      </ul>
    );
  }
}

export default connect(
  state => { return { ui: state.ui, menuItems: state.menuItem, user: state.user }},
  dispatch => bindActionCreators(Object.assign({}, uiActions, menuItemActions), dispatch)    
)(Navigation);