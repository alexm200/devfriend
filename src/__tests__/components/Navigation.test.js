import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Navigation from '../../components/navigation/Navigation';

const mockStore = configureStore();
const initialState = {
  menuItems: [
    {
        dateCreated: "2019-05-21T10:36:04.595Z",
        hasDivider: false,
        icon: "",
        isHeader: false,
        order: 1,
        text: "",
        userId: "5ce3d418848c46165880648c",
        _id: "5ce3d494848c46165880648d",
    },
    {
        dateCreated: "2019-05-21T13:06:53.489Z",
        hasDivider: false,
        icon: "",
        isHeader: false,
        order: 2,
        text: "",
        userId: "5ce3d418848c46165880648c",
        _id: "5ce3f7ed848c46165880648e",
    },
  ],
};
const store = mockStore(initialState);

describe('<Navigation />', () => {
    test('dispatches event to show the menu Items', () => {
      const wrapper = shallow(<Navigation store={store} />);
      const component = wrapper.dive();
  
      //component.find(Avatar).props().onPress();
  
      expect(store.getActions()).toMatchSnapshot();
    });
});