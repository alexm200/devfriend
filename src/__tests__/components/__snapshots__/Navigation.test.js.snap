// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> dispatches event to show the menu Items 1`] = `Array []`;
